src/go/rdctl/cmd/snapshot.go

```diff
@@ -90,37 +105,49 @@ func wrapSnapshotOperation(wrappedFunction cobraFunc) cobraFunc {
 		if err := rdClient.UpdateBackendState(desiredState); err != nil {
 			return fmt.Errorf("failed to stop backend: %w", err)
 		}
-		if err := waitForVMState(rdClient, "STOPPED"); err != nil {
+		if err := waitForVMState(rdClient, []string{"STOPPED"}); err != nil {
 			return fmt.Errorf("error waiting for backend to stop: %w", err)
 		}
 
 		functionErr := wrappedFunction(cmd, args)
+		if functionErr != nil {
+			snapshotErrors = append(snapshotErrors, functionErr)
+		}
 
 		// Start and unlock the backend
 		desiredState = client.BackendState{
 			VMState: "STARTED",
 			Locked:  false,
 		}
 		startVMErr := rdClient.UpdateBackendState(desiredState)
-		waitForStartedErr := waitForVMState(rdClient, "STARTED")
-		return errors.Join(functionErr, startVMErr, waitForStartedErr)
+		if startVMErr != nil {
+			snapshotErrors = append(snapshotErrors, startVMErr)
+		}
+
+		waitForStartedErr := waitForVMState(rdClient, []string{"STARTED", "DISABLED"})
+		if waitForStartedErr != nil {
+			snapshotErrors = append(snapshotErrors, waitForStartedErr)
+		}
+		return nil
 	}
 }
 
-func waitForVMState(rdClient client.RDClient, desiredState string) error {
+func waitForVMState(rdClient client.RDClient, desiredStates []string) error {
```

Comment:
It would be nice to have a comment on this function explaining what `desiredStates` is, now that it isn't quite as clear what this variable is.

Reply:
done